lib/internal/bridge/platform_to_dart_bridge.dart

```diff
@@ -285,6 +278,11 @@ class PlatformToDartBridge {
   }
 
   Future<void> _cancelTransactionIfExists(String transactionId) async {
-    await pendingTransactions.remove(transactionId)?.cancel();
+    await _manager.cancelMonitoringTransactionIfExists(transactionId);
+    await pendingTransactions.remove(transactionId)?.cancel()?.catchError(
```

Comment:
Perhaps pendingTransactions should be moved to SimulationManager? Some transactions are in the bridge (don't know why) and some in SimulationManager.